pkg/apis/core/validation/validation.go

```diff
@@ -3793,7 +3796,7 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	allErrs = append(allErrs, validateRestartPolicy(&spec.RestartPolicy, fldPath.Child("restartPolicy"))...)
 	allErrs = append(allErrs, validateDNSPolicy(&spec.DNSPolicy, fldPath.Child("dnsPolicy"))...)
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabels(spec.NodeSelector, fldPath.Child("nodeSelector"))...)
-	allErrs = append(allErrs, ValidatePodSecurityContext(spec.SecurityContext, spec, fldPath, fldPath.Child("securityContext"), opts)...)
+	allErrs = append(allErrs, validatePodSpecSecurityContext(spec.SecurityContext, spec, fldPath, fldPath.Child("securityContext"), opts)...)
```

Comment:
going into slow-motion skew visualization mode...

1. This relaxes validation in 1.28 by not checking hostPort values at all when validating workloads, which goes too far.
    * We want to tolerate a hostPort of 0 to allow late-defaulting, but we _don't_ want to tolerate `HostPort != 0 && HostPort != ContainerPort`.
    * I think we still need to validate host ports in all cases, but tolerate zero values when validating workloads and forbid them when validating pods. That can be plumbed as a `RequireHostPortMatchContainerPort bool` in `PodValidationOptions`, and only set to true in ValidatePodCreate/ValidatePodUpdate.

2. Normally we have to roll out validation changes over two releases, _but_ this is tolerating a zero value a 1.27 API server will default, so I _think_ we're ok
    * If a 1.28 server persists a workload with hostPort=0, a 1.27 server will default `hostPort` to match `containerPort` when reading from etcd, passing validation and avoiding a generation bump.
    * If a client reads a workload with hostPort=0 from a 1.28 server, and writes to a 1.27 server, the 1.27 server will default `hostPort` to match `containerPort` when reading from etcd, passing validation and avoiding a generation bump.

Reply:
New commit pushed with validation enhanced.